### PR TITLE
Doc fixes & sphinx update

### PR DESCRIPTION
**Pull request by gpotter2**
Content of this PR:
- Fix grammar mistakes in the doc using [Grammarly](https://www.grammarly.com)
- Update `Makefile` and `make.cmd` (replaces BuildDoc.bat), from the latest sphinx sources
- Update `conf.py` to the latest sphinx format
- _**Remove links scapy-ml mail list (needed ?)**_

Note: don't use compare mode to review the Makefile + make.cmd files, they are 87% different

### CODE REVIEW

File: doc/scapy/troubleshooting.rst [hunk 1]
```diff
@@ -62,8 +62,8 @@ Getting help
 
 Common problems are answered in the FAQ.
 
-There's a low traffic mailing list at ``scapy.ml(at)secdev.org``  (`archive <http://news.gmane.org/gmane.comp.security.scapy.general>`_, `RSS, NNTP <http://gmane.org/info.php?group=gmane.comp.security.scapy.general>`_). You are encouraged to send questions, bug reports, suggestions, ideas, cool usages of Scapy, etc. to this list. Subscribe by sending a mail to ``scapy.ml-subscribe(at)secdev.org``.
```
Comment (gpotter2) [inline doc/scapy/troubleshooting.rst]: I thought this maillist was inactive, but from what I've checked it is still working
image

Should we leave the link to it ?

Comment (guedou) [reply doc/scapy/troubleshooting.rst]: I don't use it but I don't think that we can remove it. 

Also, can you add a reference to the gitter channel ?

**Comment by guedou** [comment]
Do you think that it will still work fine with https://scapy.readthedocs.io/en/latest/ ?

**Comment by gpotter2** [comment]
I've built the doc locally, and it worked. Should be fine

File: doc/scapy/troubleshooting.rst (hunk 1 above)
Comment (gpotter2) [reply doc/scapy/troubleshooting.rst]: Sure

**Comment by guedou** [comment]
Thanks for this work!

**Pull request merged**